src/sentry/notifications/activity/base.py

```diff
@@ -34,27 +45,46 @@ def _get_subject_prefix(self) -> str:
     def should_email(self) -> bool:
         return True
 
-    def get_participants(self) -> Mapping[User, int]:
+    def get_providers_from_which_to_remove_user(
+        self,
+        user: User,
+        participants_by_provider: Mapping[ExternalProviders, Mapping[User, int]],
+    ) -> Set[ExternalProviders]:
+        """
+        Given a mapping of provider to mappings of users to why they should receive
+        notifications for an activity, return the set of providers where the user
+        has opted out of receiving notifications.
+        """
+
+        providers = {
+            provider
+            for provider, participants in participants_by_provider.items()
+            if user in participants
+        }
+        if (
+            providers
+            and UserOption.objects.get_value(user=user, key="self_notifications", default="0")
+            == "1"
+        ):
+            return providers
+        return set()
+
+    def get_participants(self) -> Mapping[ExternalProviders, Mapping[User, int]]:
         # TODO(dcramer): not used yet today except by Release's
         if not self.group:
             return {}
 
-        participants: MutableMapping[User, int] = GroupSubscription.objects.get_participants(
-            group=self.group
-        )[ExternalProviders.EMAIL]
-
-        if self.activity.user is not None and self.activity.user in participants:
-            receive_own_activity = (
-                UserOption.objects.get_value(
-                    user=self.activity.user, key="self_notifications", default="0"
-                )
-                == "1"
+        participants_by_provider = GroupSubscription.objects.get_participants(self.group)
+        user_option = self.activity.user
+        if user_option:
+            # Remove the actor that created the activity from the recipients list.
```

Comment:
what about when people opt into receiving notifications about their own activity?

Reply:
I updated the comment with a clarification.